common/djangoapps/student/tests/factories.py

```diff
@@ -154,7 +154,14 @@ def _create(cls, model_class, *args, **kwargs):
                 course_overview = CourseOverviewFactory(**course_kwargs)
             kwargs['course'] = course_overview
 
-        return manager.create(*args, **kwargs)
+        course_enrollment = manager.create(*args, **kwargs)
+
+        if course_enrollment is not None and 'created' in kwargs:
+            # Let factory add created datetime
+            course_enrollment.created = kwargs['created']
+            course_enrollment.save()
+
+        return course_enrollment
```

Comment:
This is a core change, lets comment it's needs and usage.

Reply:
This change is not required in the unit test now, so reverting it back.